src/test/java/io/vertx/openapi/contract/OpenAPIContractTest.java

```diff
@@ -112,6 +114,48 @@ void testFromFailsInvalidSpec(Vertx vertx, VertxTestContext testContext) {
     })));
   }
 
+  @Test
+  @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
+  void testValidJsonSchemaProvidedAsAdditionalSpecFiles(Vertx vertx, VertxTestContext testContext) {
+    Path resourcePath = getRelatedTestResourcePath(OpenAPIContractTest.class).resolve("split");
+    JsonObject contract = loadJson(vertx, resourcePath.resolve("petstore.json"));
+    JsonObject invalidComponents = loadJson(vertx, resourcePath.resolve("validJsonSchemaComponents.json"));
+    JsonObject validComponents = loadJson(vertx, resourcePath.resolve("components.json"));
+
+    Map<String, JsonObject> additionalValidSpecFiles = ImmutableMap.of("https://example.com/petstore", validComponents);
+    Map<String, JsonObject> additionalInvalidSpecFiles = ImmutableMap.of("https://example.com/petstore", invalidComponents);
+
+    OpenAPIContract.from(vertx, contract.copy(), additionalValidSpecFiles)
+      .compose(validResp -> Future.succeededFuture(validResp.getRawContract()))
+      .onSuccess(validJsonRef -> OpenAPIContract.from(vertx, contract.copy(), additionalInvalidSpecFiles)
+        .onSuccess(splitResp -> testContext.verify(() -> {
+          assertThat(splitResp.getRawContract()).isEqualTo(validJsonRef);
+          testContext.completeNow();
+        }))
+        .onFailure(testContext::failNow))
+      .onFailure(testContext::failNow);
+  }
+
+  @Test
+  @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
+  void testMalformedJsonSchemaProvidedAsAdditionalSpecFiles(Vertx vertx, VertxTestContext testContext) {
+    Path resourcePath = getRelatedTestResourcePath(OpenAPIContractTest.class).resolve("split");
+    JsonObject contract = loadJson(vertx, resourcePath.resolve("petstore.json"));
+    JsonObject malformedComponents = loadJson(vertx, resourcePath.resolve("malformedComponents.json"));
+
+    Map<String, JsonObject> additionalMalformedSpecFiles = ImmutableMap.of("https://example.com/petstore", malformedComponents);
+
+    OpenAPIContract.from(vertx, contract.copy(), additionalMalformedSpecFiles)
+      .onComplete(handler -> testContext.verify(() -> {
+        assertTrue(handler.failed());
+        assertThat(handler.cause()).isInstanceOf(OpenAPIContractException.class);
+        assertThat(handler.cause()).hasMessageThat()
+          .isEqualTo("The passed OpenAPI contract is invalid: Found issue in specification for reference: " +
+            "Instance type string is invalid. Expected array");
```

Comment:
In the linked stackoverflow example there is an issue with type "any". How is this error message related to this error? Do we test the correct error?

Reply:
I've removed this example to replace it with a more straight forward example.